image_source_stream.go

```diff
@@ -4,8 +4,26 @@ import (
 	"context"
 )
 
+type ErrorHandler interface {
+	// HandleError receives the error returned by ImageSource.Next
+	// regardless of whether or not the error is nil (This allows
+	// for error handling logic based on consecutively retrieved errors).
+	// It returns a boolean indicating whether or not the loop should continue.
+	HandleError(ctx context.Context, frameErr error) bool
+}
+
+type defaultErrorHandler struct{}
+
+func (eH *defaultErrorHandler) HandleError(ctx context.Context, frameErr error) bool {
+	if frameErr != nil {
+		Logger.Debugw("error getting frame", "error", frameErr)
+		return true
+	}
+	return false
+}
+
 // streamSource will stream a source of images forever to the stream until the given context tells it to cancel.
-func streamSource(ctx context.Context, once func(), is ImageSource, stream Stream) {
+func streamSource(ctx context.Context, once func(), is ImageSource, stream Stream, errHandler ErrorHandler) {
```

Comment:
just make this a function, not an interface.

Reply:
so I was going to do that, but the issue is that the current logic from the other PR was only introducing wait for _consecutive_ error counts. Only a struct could keep track of this, which is why I was opting for an interface. Is there some other way I should accommodate for this in the design?

Reply:
a function can do that with a closure

Reply:
a func returning a func

Reply:
yeah, fair enough. Will change